storyteller-api/pkg/yandexgpt/client.go

```diff
@@ -0,0 +1,154 @@
+package yandexgpt
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/rep-co/fablescope-backend/storyteller-api/pkg/yandexgpt/internal"
+)
+
+// YandexGPT Client
+type YandexGPTClient struct {
+	requestBuilder internal.RequestBuilder
+	config         *YandexGPTClientConfig
+}
+
+// Creates new YandexGPT Client.
+//
+// If you're using this option, keep in mind that you will need to generate IAM  token yourself.
+func NewYandexGPTClientWithIAMToken(
+	iamToken string,
+) *YandexGPTClient {
+	config := NewYandexGPTClientConfigWithIAMToken(iamToken)
+
+	return &YandexGPTClient{
+		config:         config,
+		requestBuilder: internal.NewRequestBuilder(),
+	}
+}
+
+// Creates new YandexGPT Client.
+func NewYandexGPTClient() *YandexGPTClient {
+	config := NewYandexGPTClientConfig()
+
+	return &YandexGPTClient{
+		config:         config,
+		requestBuilder: internal.NewRequestBuilder(),
+	}
+}
+
+// Creates new YandexGPT Client.
+//
+// You will need to specify your own API key.
+func NewYandexGPTClientWithAPIKey(
+	apiKey string,
+) *YandexGPTClient {
+	config := NewYandexGPTClientConfigWithAPIKey(apiKey)
+
+	return &YandexGPTClient{
+		config:         config,
+		requestBuilder: internal.NewRequestBuilder(),
+	}
+}
+
+func (c *YandexGPTClient) newRequest(
+	ctx context.Context,
+	method,
+	url string,
+	req YandexGPTRequest,
+) (*http.Request, error) {
+	request, err := c.requestBuilder.Build(ctx, method, url, req)
+	if err != nil {
+		return nil, err
+	}
+
+	c.setHeaders(request)
+
+	return request, nil
+}
+
+func (c *YandexGPTClient) sendRequest(
+	request *http.Request,
+	v Response,
+) error {
+	response, err := c.config.HTTPClient.Do(request)
+	if err != nil {
+		return err
+	}
+
+	defer response.Body.Close()
+
+	if internal.IsBadStatusCode(response.StatusCode) {
+		return c.handleResponseError(response)
+	}
+
+	if v != nil {
+		v.SetHeader(response.Header)
+	}
+
+	return json.NewDecoder(response.Body).Decode(v)
+}
+
+func (c *YandexGPTClient) setHeaders(request *http.Request) {
+	request.Header.Set("Content-Type", "application/json")
+
+	if c.config.ApiKey != "" {
```

Comment:
Please make ApiKey field optional

Reply:
There are no such thing as mandatory or optional field in Go. However, I understand, that having API key set as an empty string while using client with IAM token might be a little confusing. In that case, I propose a fix: using pointers to the key/token instead of an actual string. That way we can check if API key is nil.

Reply:
If you think empty string is ok, let's leave it